### PR TITLE
Fused softplus

### PR DESCRIPTION
**Pull request by Silv3S**
### PR types
Others

### PR changes
OPs

### Describe
- add `fused_softplus` op with kernel to handle activation fuse pass logic,
- remove extra attributes from base kernel,
- adjust unit test,
- rename `MKLDNN` -> `oneDNN`.

### CODE REVIEW

**Comment by YuanRisheng** [comment]
@zyfncg Please help review this pr

**Review by YuanRisheng — DISMISSED** [review]

**Review by zyfncg — DISMISSED** [review]

**Comment by Silv3S** [comment]
@Aurelius84, @jim19930609 could you please review? Your approvals are needed

**Comment by jczaja** [comment]
@xinyu-intel , @YangQun1, @wlsh1up , @bin1guo Please help with review

File: paddle/phi/backends/onednn/onednn_reuse.h [hunk 1]
```diff
@@ -1623,6 +1623,47 @@ class PoolingOneDNNHandler
...
+                        const phi::DenseTensor* x,
+                        const float beta,
+                        const std::string& fuse_activation = "",
+                        const float fuse_alpha = 0.0f,
+                        const float fuse_beta = 0.0f)
+      : OneDNNHandlerNoCachingT<T, dnnl::binary>(dev_ctx.GetEngine(),
+                                                 dev_ctx.GetPlace()) {
+    dnnl::post_ops post_ops;
+    post_ops.append_eltwise(
+        1.0f, dnnl::algorithm::eltwise_soft_relu, 0.0f, 0.0f);
```
Comment (YangQun1) [inline paddle/phi/backends/onednn/onednn_reuse.h:1639]: Here, I guess you want to split the softplus operator `y = ln(1+e^(beta*x))/beta` to the following three parts ?
- binary mul primitive: `y1 = x * beta`
- soft_relu post-op: `y2 = ln(1+e^y1)`
- linear post-ops: `y = y2 / beta`

Comment (Silv3S) [reply paddle/phi/backends/onednn/onednn_reuse.h:1639]: Our version of softplus kernel is already implemented this way. In this PR the only change is that softplus_kernel doesn't pass `fuse_activation`, `fuse_alpha` and `fuse_beta` to SoftplusOneDNNHandler and fused_softplus_kernel does. That's why AppendActivation will append another oneDNN post-op which comes from `softplus_activation_onednn_fuse_pass`.

Comment (xinyu-intel) [reply paddle/phi/backends/onednn/onednn_reuse.h:1639]: Do I understant correctly that if none of analysis passes applied, single softplus will also run into oneDNN library?

Comment (Silv3S) [reply paddle/phi/backends/onednn/onednn_reuse.h:1639]: That's right. If graph didn't have any pattern `softplus` followed by `relu`/`swish`/`relu6`/`gelu` etc. then base oneDNN softplus kernel will be used.

File: paddle/phi/kernels/fusion/onednn/fused_softplus_kernel.cc [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  auto src_memory_p = handler.AcquireSrcMemory(&x);
+  auto beta_memory_p = handler.AcquireBetaMemory(&beta);
+  std::shared_ptr<dnnl::memory> dst_memory_p = nullptr;
+  if (x.IsSharedBufferWith(*out)) {
+    dst_memory_p = src_memory_p;
+    dev_ctx.template Alloc<T>(out);
+  } else {
+    dst_memory_p = handler.AcquireDstMemory(out);
+  }
+  auto binary_p = handler.AcquireForwardPrimitive();
```
Comment (xinyu-intel) [inline paddle/phi/kernels/fusion/onednn/fused_softplus_kernel.cc]:
```suggestion
  auto softplus_p = handler.AcquireForwardPrimitive();
```

File: paddle/phi/kernels/fusion/onednn/fused_softplus_kernel.cc [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  auto& astream = OneDNNContext::tls().get_stream();
+
+  const std::unordered_map<int, dnnl::memory> args = {
+      {DNNL_ARG_SRC_0, *src_memory_p},
+      {DNNL_ARG_SRC_1, *beta_memory_p},
+      {DNNL_ARG_DST, *dst_memory_p}};
+
+  binary_p->execute(astream, args);
```
Comment (xinyu-intel) [inline paddle/phi/kernels/fusion/onednn/fused_softplus_kernel.cc]:
```suggestion
  softplus_p->execute(astream, args);
```

File: paddle/phi/kernels/fusion/onednn/fused_softplus_kernel.cc (hunk 2 above)
Comment (Silv3S) [reply paddle/phi/kernels/fusion/onednn/fused_softplus_kernel.cc]: Thanks. Changed it in both `softplus` and `fused_softplus` kernels

**Review by YuanRisheng — APPROVED** [review]

**Review by Aurelius84 — APPROVED** [review]
LGTM for data type regisgtration

**Review by zyfncg — APPROVED** [review]

**Review by xinyu-intel — APPROVED** [review]

**Pull request merged**